packages/common/src/util/Disposer.ts

```diff
@@ -0,0 +1,30 @@
+import { Disposable } from "../ide/types/ide.types";
+
+/**
+ * A class that can be used to dispose of multiple disposables at once. This is
+ * useful for managing the lifetime of multiple disposables that are created
+ * together. It ensures that if one of the disposables throws an error during
+ * disposal, the rest of the disposables will still be disposed.
+ */
+export class Disposer implements Disposable {
+  private disposables: Disposable[] = [];
+
+  constructor(...disposables: Disposable[]) {
+    this.push(...disposables);
+  }
+
+  public push(...disposables: Disposable[]) {
+    this.disposables.push(...disposables);
+  }
+
+  dispose(): void {
+    this.disposables.forEach(({ dispose }) => {
+      try {
```

Comment:
I assume that `Disposable.from` doesn't have this behavior?

Reply:
Not sure. That's a VSCode thing tho; we need this in the engine